bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/internal/SemanticsMetadataProvider.java

```diff
@@ -82,10 +82,10 @@ public SemanticsMetadataProvider(final @Reference ItemRegistry itemRegistry) {
     @Activate
     protected void activate() {
         initRelations();
+        itemRegistry.addRegistryChangeListener(this);
```

Comment:
Is moving this line up here really needed? Listening to changes should happen only after the initial item set is processed and it should be made sure that those calls only come in once the service is activated.

Reply:
I moved it because without this change there is a chance that it will not receive changes when the thread gets suspended while processing all the items.

Reply:
Iirc, we are using this pattern at many places.
To be really clean, we should then have a marker that the component is not yet activated and queue the listener changes to have them processed at a later moment. Because otherwise, it could also happen that you process a change before the activate method processed its older state. That would make the overall code pretty complex, though.

Reply:
I've moved it back. Let's fix this when it really becomes an issue. :wink: